doc/user-guide/user-guide.rst

```diff
@@ -0,0 +1,598 @@
+
+The package allow users to do that through the ``intended_usage`` parameter.
+
+By default ``intended_usage = None``, which causes the class to load all the resources from the remotes, which is useful when using the classes from Jupyter notebooks.
+
+Each class defines a specific group of "usages", for different types of operations.
+It is possible to combine usages, using bitwise operation, in case more than one operation is needed, e.g.;
+
+.. code:: python
+
+    intended_usage = Usages.StateTransition | Usages.MonitorHeartBeat
+
+There is also a ``Usages.All`` option that should include the topics for all the supported operations (not to be confused with the ``intended_usage = None``, which loads all the topics from all the components in a group).
+
+The feature is available when instantiating the class, for example;
+
+.. code:: python
+
+    from lsst.ts import salobj
+    from lsst.ts.observatory.control.auxtel import ATCS, ATCSUsages
+
+
+    class MyScript(salobj.BaseScript):
+
+      def __init__(self, index):
+
+          super().__init__(index=index, descr="My script for some operation.")
+
+          self.atcs = ATCS(domain=self.domain, log=self.log, intended_usage=ATCSUsages.StateTransition)
+
+
+Details of the available usages for each class is given furthermore.
+
+.. _user-guide-generic-csc-group-behavior:
+
+Generic CSC Group behavior
+==========================
+
+All CSC Group classes are constructed on top of the :py:class:`RemoteGroup <lsst.ts.observatory.control.RemoteGroup>` base class, which implement generic behavior for groups of CSCs.
+These generic methods and attributes will (mostly) work equally regardless of the group class.
+
+Probably the most commonly used generic operations are the :py:meth:`enable <lsst.ts.observatory.control.RemoteGroup.enable>` and :py:meth:`standby <lsst.ts.observatory.control.RemoteGroup.standby>` methods.
+
+The idea is that, after running :py:meth:`enable <lsst.ts.observatory.control.RemoteGroup.enable>`, all CSCs in the group will be in the ``ENABLED`` state.
+Components that where already in ``ENABLED`` state will be left in that state.
+For components in any other state the method will make sure to send the required state transition commands to enable them.
+The method can run without any input argument, though one must be aware that it need to know what setting to load when transition from ``STANDBY`` to ``DISABLED``.
+If no input is given, the method will inspect the ``settingVersions`` event from the CSCs in the group and select the first item in ``settingVersions.recommendedSettingsLabels``.
+It is possible to provide settings either partially of fully using the ``settings`` input parameter.
+
+Following are a couple of examples of how to use the :py:meth:`enable <lsst.ts.observatory.control.RemoteGroup.enable>` method.
+
+This will inspect the ``settingVersions`` event from all CSCs in the `ATCS` group to determine the ``settingsToApply`` for each one of them.
```

Comment:
I wish we could link to more information on configuration here.... but until we get the config stuff implemented i suppose it's not feasible. "How do I inspect the settingVersions" is the immediate question. Maybe put an example of how to print that info?

Reply:
This seems like a nice feature for the `RemoteGroup` let me add this as a new feature and document it later. I'll add a comment here to document once the feature exists.